content/blog/explore-the-hpe-greenlake-for-compute-ops-management-rest-api-using-python-and-powershell.md

```diff
@@ -0,0 +1,242 @@
+
+Once the access token is generated, it can be used to make API calls to perform any HTTP method against the desired HPE GreenLake for Compute Ops Management resource. To do this, you simply add the access token to the HTTP header with the keyword "Authorization: Bearer {token}". The name “Bearer authentication” can be understood as “give access to the bearer of this token”.
+
+There are several ways to invoke the API:
+
+* Using REST directly or from JavaScript 
+* Using cURL
+* Using Python
+* Using Go
+* Using PowerShell
+* Using Ansible, Terraform, etc.
+
+The [HPE GreenLake for Compute Ops Management API Reference site](https://developer.greenlake.hpe.com/docs/greenlake/services/compute-ops/public/openapi/compute-ops-latest/overview/) leverages an OpenAPI conformant documentation that provides a complete explanation of the operations supported by the Unique Resource Identifiers [(URIs)](<>)as well as sample requests and responses.
+
+# Using PowerShell
+
+From a Terminal application, I start a PowerShell environment and set some variables that will be used later. 
+
+```powershell
+# Store our Compute Ops Manager API endpoint
+$endpoint = "https://us-west2-api.compute.cloud.hpe.com"
+# Initialize an empty headers array
+$headers = @{} 	
+# Add a header for authorization, using the Access Token generated in task 2	
+$headers["Authorization"] = "Bearer_access_token_here"
+```
+
+![figure3](/img/greenlake-com-ops-api-pyt-pow1.png)
+
+Our first request:
+
+```powershell
+$response = Invoke-webrequest "$endpoint/compute-ops/v1beta2/servers?limit=2" -Method GET -Headers $headers
+```
+
+When the request is successful, the response should display 200:
+
+![figure4](/img/greenlake-com-ops-api-pyt-pow2.png)
+
+```powershell
+$response.StatusCode
+```
+
+The JSON data returned by the request is in $response.Content but it is rather hard to read as is, I can prettify it a bit with: 
+
+```powershell
+$response.Content | ConvertFrom-Json | Format-List
+```
+
+Now, the request returned 2 objects (count : 2) and the actual server details are in the items array property.
+
+![figure5](/img/greenlake-com-ops-api-pyt-pow3.png)
+
+The API supports several query parameters depending on the resource type, such as filter, limit (maximum number of records to return), offset (resource offset to start the response from) and sort (order in which to return the resources in the collection).
+
+Filters provide the ability to limit the resources that take part in the action of a REST call. When a REST call includes a filter, the GET or DELETE action is restricted to a response that meets the filter requirements. Filters are specified by using the query parameter 'filter'.
+
+A simple example of filtering follows:
+
+**GET <URI>?filter=powerState eq 'Off'**
+
+This example shows a simple filter. The resources returned by the query are limited to results with the attribute powerState and the value Off. 
```

Comment:
If you reword it, as I suggest above, then you don't need this first sentence in 104 and you can start line 104 with "The resources returned by the query are..."

The reason I prefer it this way is because it's always better to tell the reader up front what you're going to do as opposed to doing something and then explaining what it was. The human brain reads and understands (in general... but not in all languages) top to bottom.